activesupport/lib/active_support/core_ext/module/deprecation.rb

```diff
@@ -1,25 +1,26 @@
 # frozen_string_literal: true
 
 class Module
-  #   deprecate :foo
-  #   deprecate bar: 'message'
-  #   deprecate :foo, :bar, baz: 'warning!', qux: 'gone!'
-  #
-  # You can also use custom deprecator instance:
-  #
   #   deprecate :foo, deprecator: MyLib::Deprecator.new
   #   deprecate :foo, bar: "warning!", deprecator: MyLib::Deprecator.new
   #
-  # \Custom deprecators must respond to <tt>deprecation_warning(deprecated_method_name, message, caller_backtrace)</tt>
-  # method where you can implement your custom warning behavior.
+  # Deprecators are typically instance of ActiveSupport::Deprecation, but you can also pass any object that responds to
+  # <tt>deprecation_warning(deprecated_method_name, message, caller_backtrace)</tt> where you can implement your custom
+  # warning behavior.
   #
   #   class MyLib::Deprecator
   #     def deprecation_warning(deprecated_method_name, message, caller_backtrace = nil)
   #       message = "#{deprecated_method_name} is deprecated and will be removed from MyLibrary | #{message}"
   #       Kernel.warn message
   #     end
   #   end
-  def deprecate(*method_names)
-    ActiveSupport::Deprecation.deprecate_methods(self, *method_names)
+  def deprecate(*method_names, deprecator: nil, **options)
+    if deprecator.is_a?(ActiveSupport::Deprecation)
+      deprecator.deprecate_methods(self, *method_names, **options)
+    elsif deprecator
+      ActiveSupport::Deprecation._instance.deprecate_methods(self, *method_names, **options, deprecator: deprecator)
+    else
+      ActiveSupport.deprecator.warn("Module.deprecate without a deprecator is deprecated")
+    end
```

Comment:
```suggestion
    else
      if deprecator.nil?
        ActiveSupport.deprecator.warn("Module.deprecate without a deprecator is deprecated")
        deprecator = ActiveSupport::Deprecation._instance
      end
      ActiveSupport::Deprecation._instance.deprecate_methods(self, *method_names, **options, deprecator: deprecator)
    end
```

Shouldn't the last branch keep the old behavior (along with the warning)? Otherwise its effectively a removal instead of a deprecation

Maybe the `"deprecate_methods delegator is deprecated"` test should be updated to ensure that the deprecation is still in place as well?